toplevel/opttopmain.ml

```diff
@@ -204,11 +233,21 @@ module Options = Main_args.Make_opttop_options (struct
   let _open s = open_modules := s :: !open_modules
   let _plugin p = Compplugin.load p
 
+  let _args = wrap_expand Arg.read_arg
+  let _args0 = wrap_expand Arg.read_arg0
+
   let anonymous = file_argument
 end);;
 
 let main () =
   native_code := true;
-  Arg.parse Options.list file_argument usage;
+  let list = ref Options.list in
+  begin
+    try
+      Arg.parse_and_expand_argv_dynamic current argv list file_argument usage;
+    with
+    | Arg.Bad msg -> Format.fprintf Format.err_formatter "%s%!" msg; exit 2
+    | Arg.Help msg -> Format.fprintf Format.std_formatter "%s%!" msg; exit 0
```

Comment:
Apologies for being very slow in going through my review, but: why do we have this new exception-handling code here? The new failure mode you added does `exit 2` directly, if I understand correctly, so is there still a reason for this?